common/types.go

```diff
@@ -669,3 +1084,16 @@ func (l Location) RPCMarshal() []hexutil.Uint64 {
 
 	return res
 }
+
+func IsInChainScope(b []byte) bool {
```

Comment:
This should probably be a private helper method _only_ used by the `NewAddress()` constructor. Everywhere else in the codebase should be using type assertions on an already constructed `Address` type